code/modules/projectiles/guns/energy/special.dm

```diff
@@ -142,10 +141,12 @@
 		var/obj/item/stack/sheet/S = A
 		S.use(1)
 		power_supply.give(1000)
+		recharge_newshot(1)

```

Comment:
power cell was recharged: we try to reload the chamber if it was empty (because of lack of power).